Velaptor/OpenGL/GLWindow.cs

```diff
@@ -249,23 +283,77 @@ public async Task ShowAsync(Action? afterStart = null, Action? afterUnloadAction
         public void Dispose() => Dispose(true);
 
-        /// <summary>
-        /// Invoked when an OpenGL error occurs.
-        /// </summary>
-        private static void GL_GLError(object? sender, GLErrorEventArgs e) => throw new Exception(e.ErrorMessage);
+        private void RunGLWindow()
+        {
+            this.glWindow.Run();
+
+            /*NOTE:
+             * Only dispose of the window here and not in the Dispose() method!!
+             *
+             * This is because this line of code will not be executed until the Window.Run() method
+             * has finished executing.  This happens once the window is closed.
+             *
+             * If you dispose of the window in the Dispose() method before the Run() method is finished
+             * then the application will crash.
+             */
+            this.glWindow.Dispose();
+        }
+
+        private void PreInit()
+        {
+            this.glWindow = this.windowFactory.CreateSilkWindow();
+
+            if (this.glWindow is null)
+            {
+                throw new InvalidOperationException("The constructed native window cannot be null.");
+            }
+
+            this.glWindow.UpdatesPerSecond = 120;
+            this.glWindow.Load += GLWindow_Load;
+            this.glWindow.Closing += GLWindow_Closing;
+            this.glWindow.Resize += GLWindow_Resize;
+            this.glWindow.Update += GLWindow_Update;
+            this.glWindow.Render += GLWindow_Render;
+        }
+
+        private void Init(uint width, uint height)
+        {
+            this.glContextReactable.PushNotification(new GLContextData(this.glWindow));
+
+            this.glWindow.Size = new Vector2D<int>((int)width, (int)height);
+            this.glInputContext = this.inputFactory.CreateInput();
+
+            if (this.glInputContext.Keyboards.Count <= 0)
+            {
+                throw new NoKeyboardException("Input Exception: No connected keyboards available.");
+            }
+
+            this.glInputContext.Keyboards[0].KeyDown += GLKeyboardInput_KeyDown;
+            this.glInputContext.Keyboards[0].KeyUp += GLKeyboardInput_KeyUp;
+
+            if (this.glInputContext.Mice.Count <= 0)
+            {
+                throw new NoMouseException("Input Exception: No connected mice available.");
+            }
+
+            this.glInputContext.Mice[0].MouseDown += GLMouseInput_MouseDown;
+            this.glInputContext.Mice[0].MouseUp += GLMouseInput_MouseUp;
+            this.glInputContext.Mice[0].MouseMove += GLMouseMove_MouseMove;
+            this.glInputContext.Mice[0].Scroll += GLMouseInput_MouseScroll;
+        }
 
         /// <summary>
         /// Invokes the <see cref="Initialize"/> action property.
         /// </summary>
-        private void GameWindow_Load(object? sender, EventArgs e)
+        private void GLWindow_Load()
         {
             // OpenGL is ready to take function calls after this Init() call has ran
```

Comment:
Grammar
```suggestion
            // OpenGL is ready to take function calls after this Init() call has run
```

Reply:
Change to the suggestion below:

```suggestion
            // OpenGL is ready to take function calls after this Init() call has executed
```